src/Web/Scotty/Internal/Types.hs

```diff
@@ -0,0 +1,240 @@
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Web.Scotty.Internal.Types where
```

Comment:
* moved from ./Web/Scotty/Internal/Types.hs.
* remove CPP
* derive (functionally compatible) instances instead of writing them